### PR TITLE
Add create user validation

### PR DESCRIPTION
**Pull request by sakirdemirhan**
Validations are added during the user creation.

### CODE REVIEW

File: src/app/api/user/createUser/route.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { NextResponse } from "next/server";
+
+export async function POST(request: Request) {
+  const body = await request.json();
+  const userData = body["user"] as User;
+  let response;
+
+  try {
+    // Create user document
+    await firebaseOperations.createDocumentWithCustomId(
```
Comment (writeriks) [inline src/app/api/user/createUser/route.ts:17]: why dont we validate the user here before create?

File: src/services/user-service/user-service.ts [hunk 2]
```diff
@@ -138,16 +136,39 @@ class UserService {
...
-    }
+    const requestBody = {
+      user,
+    };
+    const result = await fetch("/api/user/createUser", {
+      method: "POST",
+      body: JSON.stringify(requestBody),
+    });
+
+    if (!result.ok) throw new Error(result.statusText);
```
Comment (writeriks) [inline src/services/user-service/user-service.ts:147]: i would say this should return a response either user or at least 200 code to make sure request succeeded.

File: src/app/api/user/validateUser/route.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import {
```
Comment (writeriks) [inline src/app/api/user/validateUser/route.ts:1]: Im not sure if we need this endpoint?

File: src/app/api/user/createUser/route.ts (hunk 1 above)
Comment (sakirdemirhan) [reply src/app/api/user/createUser/route.ts:17]: We need to validate user before adding user to Firebase users. That's why we need to separate endpoints and validate the user here: `src/services/firebase-service/firebase-auth-service.ts`

File: src/app/api/user/validateUser/route.ts (hunk 3 above)
Comment (sakirdemirhan) [reply src/app/api/user/validateUser/route.ts:1]: We need to validate user before adding user to Firebase users. That's why we need to separate endpoints and validate the user here: `src/services/firebase-service/firebase-auth-service.ts`

File: src/services/user-service/user-service.ts (hunk 2 above)
Comment (sakirdemirhan) [reply src/services/user-service/user-service.ts:147]: I think we need to return response only in the endpoints. This one is the function which is calling the endpoint. In the api/validateUser it already returns Response 200.

**Pull request merged**